### ISSUE TITLE
[Question] Is there way to add subcommand ?

### ISSUE THREAD

**Issue by js2me** [tj/commander.js#1813]
Hello, thanks for your powerful tool !   
I have program   

```
const program = new Command(name);

program
  .alias("sta")
  .version(version, "-v, --version", "output the current version")
  .description("Generate api via swagger scheme.\nSupports OA 3.0, 2.0, JSON, yaml.")
  .requiredOption("-p, --path <string>", "path/url to swagger scheme")
  .option("-o, --output <string>", "output path of typescript api file", "./")
  
const options = program.parse(process.argv).opts();
// ... my operation
```    

And program usage is `swagger-typescript-api --param1 --param2 --paramN`    

I need to add extra command like   
```
swagger-typescript-api my-super-command --other-param1 --other-param2 --other-paramN
```  

How can I do that ?

**Comment by shadowspawn** [comment]
Have a look at the README section on Commands: https://github.com/tj/commander.js#commands

You add a command, add options to the command (just like the program), and add an action handler. The action handler is the big difference between a program with and without subcommands. Instead of your code to do "my operation" going after the call to parse, your code gets called from the action handler.

In short:
```js
program.command('my-super-command')
    .option('--other-param1')
    .action((options) => console.log('my-super-command got called));
```

**Comment by js2me** [comment]
Hello @shadowspawn , thanks for your answer but I occurred with problem wuth using following code  
```

const rootProgram = new Command(name);

rootProgram
  .alias("sta")
  .version(version, "-v, --version", "output the current version")
  .description("Generate api via swagger scheme.\nSupports OA 3.0, 2.0, JSON, yaml.")
  .requiredOption("-p, --path <string>", "path/url to swagger scheme")
  .option("-o, --output <string>", "output path of typescript api file", "./")
  .option("-n, --name <string>", "name of output typescript api file", "Api.ts")
  .option("-t, --templates <string>", "path to folder containing templates")
  .option(
    "-d, --default-as-success",
    'use "default" response status code as success response too.\n' +
      'some swagger schemas use "default" response status code as success response type by default.',
    false,
  )
  .option(
    "-r, --responses",
    "generate additional information about request responses\n" + "also add typings for bad responses",
    false,
  )
  .option("--union-enums", 'generate all "enum" types as union types (T1 | T2 | TN)', false)
  .option("--add-readonly", "generate readonly properties", false)
  .option("--route-types", "generate type definitions for API routes", false)
  .option("--no-client", "do not generate an API class", true)
  .option("--enum-names-as-values", "use values in 'x-enumNames' as enum values (not only as keys)", false)
  .option(
    "--extract-request-params",
    "extract request params to data contract (Also combine path params and query params into one object)",
    false,
  )
  .option("--extract-request-body", "extract request body type to data contract", false)
  .option("--extract-response-body", "extract response body type to data contract", false)
  .option("--extract-response-error", "extract response error type to data contract", false)
  .option("--modular", "generate separated files for http client, data contracts, and routes", false)
  .option("--js", "generate js api module with declaration file", false)
  .option(
    "--module-name-index <number>",
    "determines which path index should be used for routes separation (example: GET:/fruites/getFruit -> index:0 -> moduleName -> fruites)",
    0,
  )
  .option("--module-name-first-tag", "splits routes based on the first tag", false)
  .option("--disableStrictSSL", "disabled strict SSL", false)
  .option("--disableProxy", "disabled proxy", false)
  .option("--axios", "generate axios http client", false)
  .option("--unwrap-response-data", "unwrap the data item from the response", false)
  .option("--disable-throw-on-error", "Do not throw an error when response.ok is not true", false)
  .option("--single-http-client", "Ability to send HttpClient instance to Api constructor", false)
  .option("--silent", "Output only errors to console", false)
  .option("--default-response <type>", "default type for empty response schema", "void")
  .option("--type-prefix <string>", "data contract name prefix", "")
  .option("--type-suffix <string>", "data contract name suffix", "")
  .option("--clean-output", "clean output folder before generate api. WARNING: May cause data loss", false)
  .option("--api-class-name <string>", "name of the api class")
  .option("--patch", "fix up small errors in the swagger source definition", false)
  .option("--debug", "additional information about processes inside this tool", false)
  .option("--another-array-type", "generate array types as Array<Type> (by default Type[])", false)
  .action(() => {
    //
    const options = rootProgram.parse(process.argv).opts();
  });

rootProgram
  .command("test", "test")
  .option("--extract-request-body", "extract request body type to data contract", false)
  .option("--extract-response-body", "extract response body type to data contract", false)
  .action(() => {
    console.info("kek");
  });

rootProgram.parse(process.argv);
```   

When I call `test --help` it reads all other options for the root operation   

```
> node index.js test --help

error: required option '-p, --path <string>' not specified
```

**Comment by shadowspawn** [comment]
The program options apply to all the subcommands too. When that isn't what you want you need to move the old "program" down into a subcommand beside the new subcommand.

But assuming you mostly just want to use that subcommand (the old program) you can make it the default command. Then you still don't have to type the subcommand name for the old behaviour!

See:
- https://github.com/tj/commander.js/issues/1616 
- https://github.com/tj/commander.js/issues/1750#issuecomment-1151607353

If that doesn't make enough sense, ask and I can write a wee example.

**Comment by shadowspawn** [comment]
FYI, in your recent example code you have:
```js
  .action(() => {
    //
    const options = rootProgram.parse(process.argv).opts();
  });
```

You should not be calling parse again. You actually get handed the options in the action handler like this:

```js
  .action((options) => {
    //
  });
```
